### ISSUE TITLE
semaphore_wait_trap hang with using in XCode 6 / Swift project

### ISSUE THREAD

**Issue by dethell** [lauraskelton/simple-share#2]
I successfully used the SimpleShare code in an XCode 5 project and I'm now bringing my code into a new XCode 6 / Swift project. I'm using the SimpleShare code as-is in Objective-C and importing through my bridging header.

The app hangs with a semaphore_wait_trap during init on this line:

_centralManager = [[CBCentralManager alloc] initWithDelegate:self queue:centralQueue];

Setting the queue to nil or to DISPATCH_QUEUE_CONCURRENT does not make any difference. It still hangs on init.

**Comment by lauraskelton** [comment]
I can't reproduce this issue- could you give some more information about your setup/build settings?

I created [a new demo app in Xcode 6 - Beta 4](https://github.com/lauraskelton/simple-share-swift) that uses a simple swift project with SimpleShare in Objective-C and a swift bridging header. It is working and transmits data via Bluetooth LE between several devices running iOS 7.1 to iOS 8.0.

Let me know if your build settings or development environment are different from those in this app, or if there are more steps you can share to reproduce this issue. Otherwise, see if your app works by implementing it in the same way I did in this demo app, and keep me posted on whether it fixes the issue!

https://github.com/lauraskelton/simple-share-swift

**Comment by dethell** [comment]
The problem happens in a couple of instances. First, I'm enabling my bluetooth manager in viewDidLoad of my view controller. I was able to get around it by wrapping my call in a dispatch_async call. Each instance I found where it was getting a wait trap I could get around it with a dispatch_async.

I'm using your code almost verbatim (which is working wonderfully, by the way). The only change I have compared to my iOS 6 project is calling the setup code in viewDidLoad. It would appear that doing so is causing a deadlock on the main thread. dispatch_async takes care of it.
